.github/workflows/tests.yaml

```diff
@@ -0,0 +1,43 @@
+name: Tests
+
+on:
```

Comment:
Is this not going to run on other branches without PR to master? Sometimes it's useful to see build of WIP branches. Is this because of some build time quota?

Reply:
Yeah, that's right, but I think if you have a draft PR open then this will trigger the build, and the build will be rerun each time the PR is updated, so you can see the results of the WIP.

TBH, this is just copied from another library where someone set things up in this way (possibly me) and it seemed sensible. Happy to change if you think worthwhile.

Reply:
Okay, thanks. I think it's fine, CI on draft PRs seems sufficient. If it turns out we need it in some nonstandard workflows like PRs targetting non-master branches, we can change it then.

Reply:
Yeah, this setup seems pretty common, and likely good baseline to start with unless proven otherwise.

One difference with tests on pushes enabled, is that once you do a PR it usually already run the tests in the background by that time since your push, so a less waiting to see any results (mostly from the PR author's point of view).